src/backend/joanie/core/api/admin.py

```diff
@@ -445,6 +446,39 @@ def destroy(self, request, *args, **kwargs):
         order.cancel()
         return Response(status=HTTPStatus.NO_CONTENT)
 
+    @extend_schema(
+        request=None,
+        responses={
+            (201, "application/json"): serializers.AdminCertificateSerializer,
+            422: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(methods=["POST"], detail=True)
+    def generate_certificate(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Generate the certificate of an order when it is eligible.
+        """
+        order = self.get_object()
+
+        certificate, created = order.get_or_generate_certificate()
+
+        if not certificate:
+            return JsonResponse(
+                {"details": "Order does not deliver certificate"},
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+        if not created:
+            return JsonResponse(
+                {"details": "The certificate already exists."},
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
```

Comment:
I feel a `422` status code is not appropriate here.
I wonder if we should return a `200` with the certificate or a `409` conflict.

Reply:
Since the endpoint is to use with a 'POST' method, I think it would suitable to return a 409 conflict.

I would return a 200 response status code with the certificate object if the endpoint was to use with the 'GET' method.

Why do I prefer the [409 status code](https://datatracker.ietf.org/doc/html/rfc7231.html#section-6.5.8) :

> The 409 (Conflict) status code indicates that the request could not
   be completed due to a conflict with the current state of the target
   resource.
   
 The conflict in this case is that the generation could not be completed due that the certificate already exists. The error message seems clear enough to my taste. What's your thought about it ?

Reply:
After discussion, we will retain the 200 status code with the objet in return.